modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/src/ModuleProcess.cs

```diff
@@ -119,12 +121,18 @@ public class ModuleProcess : IProcessControl {
                         logger.Error(ex, "Error during module execution - restarting!");
                     }
                     finally {
-                        kPublisherModuleStart.WithLabels(
-                            identity.DeviceId ?? "", identity.ModuleId ?? "").Set(0);
                         await workerSupervisor.StopAsync();
+                        if (hostScope.Resolve<ISessionManager>() is IDisposable sessionManager){
```

Comment:
Container will call dispose on both.  If the intend is to call dispose earlier, then it will be called twice which is not good.   I would remove this or we should fix the lifetime scoping.

Reply:
Dispose is not being called by the container. Both are singletons and this is intended here. The only way I could clean them up them was by manually call Dispose.

Reply:
I moved the cleanup in a StopAsync() method. The Dispose issue with singletons still remains